src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,90 @@
+import { useState, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getPeople } from '../api';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Person } from '../types';
+import { getFilteredPeople } from '../utils/getFilteredPeople';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const [searchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const peopleWithParents = people.reduce((newPeople: Person[], person) => {
+    const mother = people.find(mom => mom.name === person.motherName);
+    const father = people.find(dad => dad.name === person.fatherName);
```

Comment:
The logic for finding a person's mother and father assumes that names are unique identifiers. If there are multiple people with the same name, this could lead to incorrect parent assignments. Consider using a unique identifier, such as an ID, to find parents accurately.